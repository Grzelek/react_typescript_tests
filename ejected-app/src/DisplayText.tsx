import React, {useState} from "react";

const DisplayText = () => {
    const [txt, setTxt] = useState("")
    const [msg, setMsg] = useState("")
    const onChangeTxt = (e: React.ChangeEvent<HTMLInputElement>) =>{
        setTxt(e.target.value);
    }

    const onClickShowMsg = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) =>{
        e.preventDefault();
    
        setMsg(`witaj na tescie reacta, ${txt}!`)
    }
    return (
        <form>
            <div>
                <label>jak masz na imie chuju?</label>
            </div>
            <div>
                <input data-testid="user-in" value={txt} onChange={onChangeTxt} />
            </div>
            <div>
                <button data-testid="input-submit" onClick={onClickShowMsg}>wyswietl komunikat</button>
            </div>
            <div>
                <label data-testid="final-msg">{msg}</label>
            </div>
        </form>
    )
}

export default DisplayText

