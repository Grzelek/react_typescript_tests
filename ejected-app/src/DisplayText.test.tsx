import React from "react"
import {render, fireEvent, cleanup, waitFor} from "@testing-library/react"
import DisplayText from "./DisplayText"
import "@testing-library/jest-dom/extend-expect"
import { getValue } from "@testing-library/user-event/dist/utils"

describe("test component DisplayText", ()=>{
    const userFullName = "Jack Tester";
    const getUserFullnameMock = (
        username: string): [Promise<string>, jest.Mock<Promise<string>, [string]>] => {
            const promise = new Promise<string>((res, rej) => {
                res(userFullName)
            });
            const getUserFullname = jest.fn(
                async (username: string): Promise<string> =>{
                    return promise;
                }
            )
            return [promise, getUserFullname]
        }
       
    it("no any issue",()=>{
        const username = "testuser"
        const [promise, getUserFullname] = getUserFullnameMock(username)
        const{baseElement} = render(<DisplayText getUserFullname={getUserFullname}/>)
        //console.log(baseElement.innerHTML)
        expect(baseElement).toBeInTheDocument()
    })
     
    it("recived text from field", ()=>{
        const username = "testuser";
        const [promise, getUserFullname] = getUserFullnameMock(username)
        const {getByTestId} = render(<DisplayText getUserFullname={getUserFullname} />)
        const input = getByTestId("user-in");
        fireEvent.change(input, {target: {value:username} })
        expect(input).toBeInTheDocument();
        expect(input).toHaveValue(username)
    })
    
    it("displayed welcome text", async ()=>{
        const username = "testuser";
        const [promise, getUserFullname] = getUserFullnameMock(username)

        const msg = `hello in react test, ${userFullName}!`;
        const {getByTestId} = render(<DisplayText getUserFullname={getUserFullname}  />)
        const input = getByTestId("user-in");
        const label = getByTestId("final-msg");
        fireEvent.change(input, {target: {value:username} })
        const btn = getByTestId("input-submit")
        fireEvent.click(btn)
        
        expect(label).toMatchSnapshot();
        await waitFor(() => promise);
        expect(label.innerHTML).toBe(msg)
    })
    /*
    it("jest zgodny z migawka", async() => {
        const username = "testuser";
        const [promise, getUserFullname] = getUserFullnameMock(username)

        const msg = `hello in react test, ${userFullName}!`;
        const {getByTestId} = render(<DisplayText getUserFullname={getUserFullname}  />)
        const input = getByTestId("user-in");
        const label = getByTestId("final-msg");
        fireEvent.change(input, {target: {value:username} })
        const btn = getByTestId("input-submit")
        fireEvent.click(btn)
        
        expect(label).toMatchSnapshot();
        await waitFor(() => promise);
        expect(label.innerHTML).toBe(msg)
    })
    */
    
})

