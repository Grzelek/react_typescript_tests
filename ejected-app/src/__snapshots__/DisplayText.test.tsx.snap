// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component DisplayText displayed welcome text 1`] = `
<label
  data-testid="final-msg"
/>
`;

exports[`test component DisplayText jest zgodny z migawka 1`] = `
<label
  data-testid="final-msg"
/>
`;
